File: exercise/java/day15/src/main/java/santaChristmasList/operations/dependencies/WishList.java
package santaChristmasList.operations.dependencies;

import santaChristmasList.operations.models.Child;
import santaChristmasList.operations.models.Gift;

import java.util.HashMap;

public class WishList extends HashMap<Child, Gift> {
    public Gift identifyGift(Child child) {
        return this.getOrDefault(child, null);
    }
}

File: exercise/php/day15/src/SantaChristmasList/Dependencies/WishList.php
<?php

namespace SantaChristmasList\Dependencies;


use SantaChristmasList\Models\Child;
use SantaChristmasList\Models\Gift;

class WishList extends \ArrayObject
{
    public function identifyGift(Child $child): ?Gift
    {
        return $this->offsetExists($child->getName()) ? $this->offsetGet($child->getName()) : null;
    }

    public function set(Child $child, Gift $gift): void
    {
        $this->offsetSet($child->getName(), $gift);
    }
}